src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/INodeWithDebugInfo.cs

```diff
@@ -19,12 +23,36 @@ public DebugLocInfo(int nativeOffset, string fileName, int lineNumber, int colNu
             ColNumber = colNumber;
         }
     }
-
+    
     public interface INodeWithDebugInfo
     {
         DebugLocInfo[] DebugLocInfos
         {
             get;
         }
+
+        DebugVarInfo[] DebugVarInfos
+        {
+            get;
+        }
+    }
+
+     
+    public struct DebugVarInfo
+    {
+        public uint varNumber;
+        public string name;
+        public uint typeIndex;
+        public bool isParam;
+        public List<NativeVarInfo> ranges;
+
+        public DebugVarInfo(uint varNumber, string name, bool isParam)
+        {
+            this.name = name;
+            this.varNumber = varNumber;
+            this.typeIndex = 0x0074;
```

Comment:
Need a comment what currently you assumed. And also a link about type support that will follow.